crates/linker/src/pe.rs

```diff
@@ -1228,176 +1249,7 @@ fn write_section_header(
     data[section_header_start..][..header_array.len()].copy_from_slice(&header_array);
 }
 
-struct BaseRelocations {
-    new_size: u32,
-    delta: u32,
-}
```

Comment:
It turns out that my base relocations were totally pointless. Relocations of type 0 (absolute)` are just ignored. It makes sense that I went down this road, because what I observed was that calls to roc app functions were not relocated, and so they just jumped to invalid memory.

The fix ultimately seems to be that the base relocation directory needs to be of the correct size. I still don't fully know what the exact conditions are, but bumping it to the file alignment is something that I've seen other tools do (clang), and it makes nqueens work without regressing other examples.